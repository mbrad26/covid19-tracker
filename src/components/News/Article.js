import React from 'react'
import { Card } from 'react-bootstrap';

const Article = ({ article }) => {

  return (
      <Card id='article'>
        <Card.Img variant="top" src={article.urlToImage} />
        <Card.Body>
          <Card.Title>{article.title}</Card.Title>
          <Card.Text id='source'><span>{article.source.name}</span></Card.Text>
          <Card.Text className='text-muted far fa-clock'>
            <span> {new Date(article.publishedAt).toDateString()}</span>
          </Card.Text>
          <Card.Text id='article-description'>{article.description}</Card.Text>
          <a href={article.url} target='_blank' rel='noopener noreferrer'>Read full article ...</a>
        </Card.Body>
      </Card>
  );
};

export default Article;
