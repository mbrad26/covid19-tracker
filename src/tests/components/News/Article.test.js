import React from 'react';
import { render, screen } from '@testing-library/react';

import { article } from '../../fixtures';
import Article from '../../../components/News/Article';

describe('Article', () => {
  it('renders a snapshot', () => {
    const { container } = render(<Article article={article} />);

    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders the correct props', () => {
    render(<Article article={article} />);

    screen.debug();

    // expect(container.firstChild).toMatchSnapshot();
  });
});
