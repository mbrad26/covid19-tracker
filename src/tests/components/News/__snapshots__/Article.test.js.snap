// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article renders a snapshot 1`] = `
<div
  class="card"
  id="article"
>
  <img
    class="card-img-top"
    loading="lazy"
    src="https://techcrunch.com/wp-content/uploads/2019/06/GettyImages-1050523528.jpg?w=600"
  />
  <div
    class="card-body"
  >
    <div
      class="card-title h5"
    >
      Casa pivots to provide self-custody services to secure bitcoin
    </div>
    <p
      class="card-text"
      id="source"
    >
      <span>
        TechCrunch
      </span>
    </p>
    <p
      class="text-muted far fa-clock card-text"
    >
      <span>
         
        Thu Aug 06 2020
      </span>
    </p>
    <p
      class="card-text"
      id="article-description"
    >
      Casa, a Colorado-based provider of bitcoin security services, is launching a managed service allowing customers to buy and hold their own bitcoin, rather than using an external custodian like Coinbase. “With self-custody using Casa it’s impossible to be hacke…
    </p>
    <a
      href="http://techcrunch.com/2020/08/06/casa-pivots-to-provide-self-custody-services-to-secure-bitcoin/"
      rel="noopener noreferrer"
      target="_blank"
    >
      Read full article ...
    </a>
  </div>
</div>
`;
